### ISSUE TITLE
Don't use "dotnet run" in bootstrapper

### ISSUE THREAD

**Issue by Thealexbarney** [cake-build/frosting#32]
Using `dotnet run` results in the application using assemblies from the NuGet cache, locking the assemblies that it needs to run. MSBuild creates hard links to assemblies in the NuGet cache. This can create problems with tasks such as cleaning or rebuilding a project. Cake will be trying to delete hard links to the files that it's using to run. For some reason Windows won't let you delete a hard link to a file when another one of its hard links is locked, causing the build to crash and burn.

The solution would be to `dotnet publish` the build project and then run that published application.

This is somewhat touched on in the [official docs](https://docs.microsoft.com/en-us/dotnet/articles/core/tools/dotnet-run):

> In order to run the application, the `dotnet run` command resolves the dependencies of the application that are outside of the shared runtime from the NuGet cache. Because it uses cached dependencies, it's not recommended to use `dotnet run` to run applications in production. Instead, create a deployment using the `dotnet publish` command and deploy the published output.

**Comment by patriksvensson** [comment]
@Thealexbarney Sorry for the late reply to this. I will take a look at it as soon as I have some spare time (unless you want to take a stab at it).

**Comment by Thealexbarney** [comment]
@patriksvensson I've been doing [this](https://github.com/Thealexbarney/VGAudio/blob/e44e442777aefbd191bc1883c680eca495aab88b/build.ps1#L109-L112) in my bootstrapper. It's not pretty, but it works.

On the topic of bootstrapper changes you might be interested in, I've made it so a .NET CLI version can be specified as a variable. The script will look in the default Program Files directory, the default user install directory that dotnet-install uses, and finally the tools directory Cake uses for the local build. The CLI will be downloaded only if the specified version is not found in any of those directories. [Link here](https://github.com/Thealexbarney/VGAudio/blob/e44e442777aefbd191bc1883c680eca495aab88b/build.ps1#L47-L81)

I also removed downloading nuget in the bootstrapper so Frosting will run on CoreCLR only and not require Mono on non-windows platforms. Although my solution for downloading tools from NuGet is [somewhat](https://github.com/Thealexbarney/VGAudio/blob/e44e442777aefbd191bc1883c680eca495aab88b/build/Tasks/Restore.cs#L32-L41) [hacky](https://github.com/Thealexbarney/VGAudio/blob/e44e442777aefbd191bc1883c680eca495aab88b/build/Tools.xml).
